Makefile

```diff
@@ -31,7 +31,9 @@ OFFLINE_BUILD       ?= false
 GIT_UPSTREAM_REMOTE ?= $(shell git remote -v |grep --color=never ':Maistra/istio-operator\.git.*(fetch)' |grep --color=never -o '^[^[:space:]]*')
 
 ifeq "${GIT_UPSTREAM_REMOTE}" ""
-$(error Could not find git remote for Maistra/istio-operator)
+$(warning Could not find git remote for Maistra/istio-operator. Adding default one)
+$(shell git remote add upstream git@github.com:Maistra/istio-operator.git)
```

Comment:
Should this check to make sure `upstream` doesn't exist already?

That said, this is primarily used to update the legacy charts (`git checkout ${GIT_UPSTREAM_REMOTE}/maistra-1.0 -- resources/helm/v1.0`), so if all we want to do is compile/test, there is an `OFFLINE_BUILD` option, which maybe we could use to prevent git operations.

Reply:
Good catch. We could check, or assign another name for that - e.g. `circle_ci_upstream`.

Another option would be to move that logic to targets that rely on this value and not making it a global pre-requisite. WDYT?

Reply:
All of those sound good.

In addition, I think the update-1.0-remote target should fail in offline mode, which means the upstream remote logic could just be moved into that target.  This probably makes the most sense, i.e. don't error out just because it's not set, but only error out the target's that actually use it.

FWIW, I implemented that as a global so folks could override it via command line.  That said, I don't know if it's necessary, given the auto-setting.

Reply:
Do you want me to do that in this PR or open a new one?